packages/uxpin-merge-cli/src/steps/serialization/component/implementation/typescript/property/type/checker.ts

```diff
@@ -0,0 +1,57 @@
+import * as ts from 'typescript';
+
+import { KNOWN_TYPES_MAP } from './node/serializeKnownPropertyType';
+
+export function isAny(type: ts.Type): boolean {
+  return Boolean(type.flags & ts.TypeFlags.Any);
+}
+
+export function isBooleanLike(type: ts.Type): boolean {
+  return Boolean(type.flags & ts.TypeFlags.BooleanLike);
+}
+
+export function isCallable(type: ts.Type): boolean {
+  return isObjectLike(type) && !!type.getCallSignatures().length;
+}
+
+export function isEnum(type: ts.Type): type is ts.LiteralType {
+  if (type.symbol?.valueDeclaration) {
+    return ts.isEnumMember(type.symbol.valueDeclaration);
+  }
+
+  return false;
+}
+
+export function isKnownPropertyType(type: ts.Type): boolean {
+  const typeSymbol: ts.Symbol = type.symbol || type.aliasSymbol;
+
+  return (
+    (isAny(type) || isObjectLike(type) || isUnion(type)) &&
```

Comment:
So the trick is here, I added `isAny(type)` but I don't know why we had this condition `isObjectLike(type) || isUnion(type)` before checking it's a known type.

Reply:
When you say `but I don't know why we had this condition isObjectLike(type) || isUnion(type) before checking it's a known type.` does it refer to this part of the condition? `&& typeSymbol && typeSymbol.escapedName.toString() in KNOWN_TYPES_MAP`?

Reply:
We have a map of the known types, that maps keys detected by the TS compiler (`Array`, `Date`...) with the output we generate for each propery:

```js
export const KNOWN_TYPES_MAP: { [typeName: string]: PropertyType } = {
  Array: { name: 'array', structure: {} },
  Date: { name: 'date', structure: {} },
  ReactElement: { name: 'element', structure: {} },
  ReactNode: { name: 'node', structure: {} },
  ReactChild: { name: 'node', structure: {} },
};
```

But we used to check this map only if `isObjectLike(type) || isUnion(type)`, and with our `ReactChild` the condition is not met, it was not an object or an union, it was an `any` type, this is why I added `isAny` in the condition.

What I mean is that I don't understand why we need this condition, why not just remove it and always check if the key is in the map?

I can try but I chose to play safe, at first.